dpnp/backend/kernels/dpnp_krnl_elemwise.cpp

```diff
@@ -371,45 +387,120 @@ static void func_map_init_elemwise_1arg_1type(func_map_t& fmap)
+        get_shape_offsets_inkernel(input1_shape_data, input1_ndim, input1_shape_offsets);                              \
+        bool use_strides = !array_equal(input1_strides_data, input1_ndim, input1_shape_offsets, input1_ndim);          \
+        dpnp_memory_free_c(input1_shape_offsets);                                                                      \
                                                                                                                        \
-        DPNPC_id<_DataType_input2>* input2_it;                                                                         \
-        const size_t input2_it_size_in_bytes = sizeof(DPNPC_id<_DataType_input2>);                                     \
-        input2_it = reinterpret_cast<DPNPC_id<_DataType_input2>*>(dpnp_memory_alloc_c(input2_it_size_in_bytes));       \
-        new (input2_it) DPNPC_id<_DataType_input2>(input2_data, input2_shape, input2_shape_ndim);                      \
+        const size_t input2_shape_size_in_bytes = input2_ndim * sizeof(size_t);                                        \
+        size_t* input2_shape_offsets = reinterpret_cast<size_t*>(dpnp_memory_alloc_c(input2_shape_size_in_bytes));     \
+        get_shape_offsets_inkernel(input2_shape_data, input2_ndim, input2_shape_offsets);                              \
+        use_strides = use_strides || !array_equal(input2_strides_data, input2_ndim, input2_shape_offsets, input2_ndim);\
+        dpnp_memory_free_c(input2_shape_offsets);                                                                      \
                                                                                                                        \
-        input2_it->broadcast_to_shape(result_shape);                                                                   \
+        cl::sycl::event event;                                                                                         \
+        cl::sycl::range<1> gws(result_size);                                                                           \
                                                                                                                        \
-        const size_t result_size = input1_it->get_output_size();                                                       \
+        if (use_broadcasting)                                                                                          \
+        {                                                                                                              \
+            DPNPC_id<_DataType_input1>* input1_it;                                                                     \
+            const size_t input1_it_size_in_bytes = sizeof(DPNPC_id<_DataType_input1>);                                 \
+            input1_it = reinterpret_cast<DPNPC_id<_DataType_input1>*>(dpnp_memory_alloc_c(input1_it_size_in_bytes));   \
```

Comment:
Could you clarify where this memory is released?

Reply:
https://github.com/IntelPython/dpnp/pull/985/commits/d515febccb3171024f5fc350854c5db0f459aa80